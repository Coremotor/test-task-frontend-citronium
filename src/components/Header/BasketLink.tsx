import React from "react";
import {createUseStyles} from "react-jss";
import {useSelector} from "react-redux";
import {IStore} from "../../interfaces/interfaces";
import {NavLink} from "react-router-dom";
import {reduceProductInBasketQuantity} from "../../libs/reduceProductInBasketProps";

const useStyles = createUseStyles({
    basketLinkWrapper: {
        display: "flex",
        alignItems: "center",
        justifyContent: "space-between",
    },
    link: {
        width: 24,
        height: 24,
        marginRight: 20,
    }
})

const BasketLink = () => {

    const classes = useStyles()
    const basket = useSelector((state: IStore) => state.basket)

    const productsInBasketQuantity = reduceProductInBasketQuantity(basket)

    return (
        <div className={classes.basketLinkWrapper}>
            <NavLink className={classes.link} to="/basket">
                <img src="./img/basketIcon.svg" alt="Go to bucket"/>
            </NavLink>

            {
                productsInBasketQuantity !== 0 &&
                <span>
                {
                    productsInBasketQuantity > 9 ? '9+' : productsInBasketQuantity
                }
                </span>
            }
        </div>
    )
}

export default BasketLink;
