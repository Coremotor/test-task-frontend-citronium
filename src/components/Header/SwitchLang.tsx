import React from "react";
import {createUseStyles} from "react-jss";



const useStyles = createUseStyles({
    switchLangWrapper: {
        marginRight: 80,
    }
})

const SwitchLang = () => {

    const classes = useStyles()

    return (
        <div className={classes.switchLangWrapper}>
            <button>ru</button>
            <button>en</button>
        </div>
    )
}

export default SwitchLang;
