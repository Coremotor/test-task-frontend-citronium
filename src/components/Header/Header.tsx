import React from "react";
import SwitchLang from "./SwitchLang";
import BasketLink from "./BasketLink";
import {createUseStyles} from 'react-jss'
import {NavLink} from "react-router-dom";

const useStyles = createUseStyles({
    header: {
        color: "white",
        backgroundColor: "#094d74",
        paddingTop: 20,
        paddingBottom: 20,
    },
    headerInner: {
        display: "flex",
        alignItems: "center",
        maxWidth: 1280,
        padding: {
            right: 10,
            left: 10
        },
        margin: {
            top: 0,
            right: "auto",
            bottom: 0,
            left: "auto",
        },
    },
    headerTitleLink: {
        textDecoration: "none",
        color: "white",
        paddingLeft: 10,
        marginRight: "auto",
        marginTop: 8,
        '&:active': {
            color: "white"
        }
    },
    headerImg: {
        height: 40,
    }
})

const Header = () => {

    const classes = useStyles()
    return (
        <header className={classes.header}>
            <div className={classes.headerInner}>
                <NavLink to='/' className={classes.headerTitleLink}>
                    <img className={classes.headerImg} src="https://citronium.ru/wp-content/uploads/2019/01/Resurs-12.png" alt="Лого"/>
                </NavLink>
                <SwitchLang/>
                <BasketLink/>
            </div>
        </header>
    )
}

export default Header;
