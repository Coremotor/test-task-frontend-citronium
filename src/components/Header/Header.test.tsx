import React from "react";
import {shallow} from "enzyme";
import Header from './Header'

describe('Header', () => {
    const HeaderContainer = shallow(<Header/>)
    it('should components render', function () {

        console.log(HeaderContainer.debug())

        expect(HeaderContainer.find('header')).toHaveLength(1)
        expect(HeaderContainer.find('NavLink')).toHaveLength(1)
        expect(HeaderContainer.find('img')).toHaveLength(1)
        expect(HeaderContainer.find('SwitchLang')).toHaveLength(1)
        expect(HeaderContainer.find('BasketLink')).toHaveLength(1)
    })

    it('should equal Snapshot', function () {
        expect(HeaderContainer).toMatchSnapshot()
    })
})
