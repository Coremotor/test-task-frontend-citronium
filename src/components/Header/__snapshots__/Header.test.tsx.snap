// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should  1`] = `
<header
  className="header-0-2-1"
>
  <div
    className="headerInner-0-2-2"
  >
    <NavLink
      className="headerTitleLink-0-2-3"
      to="/"
    >
      <img
        alt="Лого"
        className="headerImg-0-2-4"
        src="https://citronium.ru/wp-content/uploads/2019/01/Resurs-12.png"
      />
    </NavLink>
    <SwitchLang />
    <BasketLink />
  </div>
</header>
`;

exports[`Header should equal Snapshot 1`] = `
<header
  className="header-0-2-1"
>
  <div
    className="headerInner-0-2-2"
  >
    <NavLink
      className="headerTitleLink-0-2-3"
      to="/"
    >
      <img
        alt="Лого"
        className="headerImg-0-2-4"
        src="https://citronium.ru/wp-content/uploads/2019/01/Resurs-12.png"
      />
    </NavLink>
    <SwitchLang />
    <BasketLink />
  </div>
</header>
`;
